aea/cli/config.py

```diff
@@ -54,89 +63,269 @@ def get(ctx: Context, json_path: List[str]):
+        except AEAException as e:
+            raise click.ClickException(*e.args)
+
+    @property
+    def attr_name(self) -> str:
+        """Attribute name."""
+        return self.json_path[-1]
+
+    def get(self) -> Any:
+        """Get config value."""
+        if self.component_id:
+            return self._get_component_value()
+        else:
+            return self._get_agent_value()
 
-def _set_config(ctx: Context, json_path: List[str], value: str, type_str: str) -> None:
-    config_loader = cast(ConfigLoader, ctx.config.get("configuration_loader"))
-    configuration_file_path = cast(str, ctx.config.get("configuration_file_path"))
+    def _get_agent_value(self) -> Any:
+        """Get config value for agent config."""
+        configuration_object = self._load_configuration_object()
+        return self._get_value_from_configuration_object(configuration_object)
 
-    configuration_object = load_yaml(configuration_file_path)
-    config_loader.validator.validate(instance=configuration_object)
+    def _get_component_value(self) -> Any:
+        """Get config value for component section in agent config or component package."""
+        configuration_object_from_agent = self._get_configuration_object_from_agent()
+        try:
+            if not configuration_object_from_agent:
+                raise click.ClickException("")
+            return self._get_value_from_configuration_object(
+                configuration_object_from_agent
+            )
+        except click.ClickException:
+            configuration_object = self._load_configuration_object()
+            return self._get_value_from_configuration_object(configuration_object)
 
-    parent_object_path = json_path[:-1]
-    attribute_name = json_path[-1]
-    parent_object = _get_and_validate_parent_obj(
-        configuration_object, parent_object_path, attribute_name
-    )
+    @property
+    def is_target_agent(self) -> bool:
+        """Is target of get/update is agent config."""
+        return self.component_id is None
 
-    type_ = FROM_STRING_TO_TYPE[type_str]
-    try:
-        if type_ != bool:
-            parent_object[attribute_name] = type_(value)
+    def _load_configuration_object(self) -> Dict:
+        """Load configuration object for component/agent."""
+        if self.is_target_agent:
+            configuration_object = self.agent_config.json
         else:
-            parent_object[attribute_name] = value not in FALSE_EQUIVALENTS
-    except ValueError:  # pragma: no cover
-        raise click.ClickException("Cannot convert {} to type {}".format(value, type_))
+            configuration_object = load_yaml(str(self.path_to_resource_configuration))
 
-    try:
-        configuration_obj = config_loader.configuration_class.from_json(
-            configuration_object
+        self.config_loader.validate(configuration_object)
+        return configuration_object
+
+    def _get_configuration_object_from_agent(self) -> Optional[Dict]:
```

Comment:
dict needs to be fully typed

Reply:
type can not be specified, configurations differs